aspnet/fundamentals/dependency-injection.rst

```diff
@@ -1,10 +1,277 @@
-.. _dependency-injection:
+Dependency Injection
```

Comment:
Generally leaves reader thinking DI is about the `Startup` class and database contexts. Need to describe or reference other uses of DI e.g. placing controllers, filters, or view location expanders in container for MVC scenarios, consuming framework services in custom middleware implementations, and adding options to affect the `AuthenticationMiddleware`. (Options are found in the container.)